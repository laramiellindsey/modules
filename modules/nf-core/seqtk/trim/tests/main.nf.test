nextflow_process {

    name "Test Process SEQTK_TRIM"
    script "../main.nf"
    process "SEQTK_TRIM"
    tag "seqtk"
    tag "seqtk/trim"
    tag "modules_nfcore"

    test("Single-end") {

        when {
            params {
                outdir = $outputDir
            }
            process {
                """
                input[0] = [ 
                    [ id:'test', single_end:true ], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll (
            { assert process.success },
            { assert snapshot(process.out).match()}
            )
        }

    }

test("Paired-end") {

        when {
            params {
                outdir = $outputDir
            }
            process {
                """
                input[0] = [ 
                    [ id:'test', single_end:false ], // meta map
                    [ 
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) 
                    ]
                ]
                """
            }
        }

        then {
            assertAll (
            { assert process.success },
            { assert snapshot(process.out).match()}
            )
        }

    }

}
