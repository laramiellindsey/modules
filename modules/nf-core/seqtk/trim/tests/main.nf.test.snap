{
    "Single-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_test_1.fastq.gz:md5,875863b402f67403dac63ef59b9c9a8a"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d061ca0231d089b087e22d2001cd7c32"
                ],
                "fastq": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_test_1.fastq.gz:md5,875863b402f67403dac63ef59b9c9a8a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d061ca0231d089b087e22d2001cd7c32"
                ]
            }
        ],
        "timestamp": "2023-10-16T10:49:29.769029"
    },
    "Paired-end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_test_1.fastq.gz:md5,875863b402f67403dac63ef59b9c9a8a",
                            "test_test_2.fastq.gz:md5,375aeb74819ca3d72203135ac80df78c"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,d061ca0231d089b087e22d2001cd7c32"
                ],
                "fastq": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test_test_1.fastq.gz:md5,875863b402f67403dac63ef59b9c9a8a",
                            "test_test_2.fastq.gz:md5,375aeb74819ca3d72203135ac80df78c"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d061ca0231d089b087e22d2001cd7c32"
                ]
            }
        ],
        "timestamp": "2023-10-16T10:49:37.792338"
    }
}